service/model/user.go

```diff
@@ -0,0 +1,25 @@
+package model
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type UserRepository interface {
+	ReadForToken(ctx context.Context, token string) (User, error)
+	Read(ctx context.Context, id uuid.UUID) (User, error)
+	ReadAll(ctx context.Context) ([]User, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+type User struct {
+	ID        uuid.UUID
+	Email     string
+	Name      string
+	AvatarUrl string
+	IsAdmin   bool
```

Comment:
I'd prefer `UserRole` custom type here, it's more flexible, but you can do that in a separate PR.